packages/vite-plugin-svelte/src/utils/options.js

```diff
@@ -579,6 +572,11 @@ function buildExtraConfigForSvelte(config) {
 	if (!isDepExternaled('svelte', config.ssr?.external ?? [])) {
 		noExternal.push('svelte', /^svelte\//);
 	}
+	// esm-env needs to be bundled by default for the development/production condition
+	// be properly used by svelte
+	if (!isDepExternaled('esm-env', config.ssr?.external ?? [])) {
+		noExternal.push('esm-env');
+	}
```

Comment:
add warning log that externalizing it will cause trouble?

(with filterable code)

Reply:
I'd personally defer this to later if someone stumbles on the issue 🤔

Reply:
ok, lets hope noone externalizes it.